src/test/java/cat/udl/eps/entsoftarch/webingogeiapi/steps/EditGameStepDefs.java

```diff
@@ -0,0 +1,212 @@
+        stepDefs.result = stepDefs.mockMvc.perform(
+                get("/games/{id}", id)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+
+    @Given("^There is a game with the pricePerCard (\\d+.\\d+) and id (\\d+)$")
+    public void thereIsAGameWithThePricePerCardAndId(double pricePerCard, int id) throws Exception {
+        JSONObject game = new JSONObject();
+        game.put("pricePerCard",pricePerCard);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                post("/games/{id}",id)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(game.toString())
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+
+
+    @When("^I edit the game with id (\\d+)$")
+    public void iEditTheGameWithId(int id) throws Exception {
+        JSONObject game = new JSONObject();
+        game.put("id", id);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                put("/games/{id}",id)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(game.toString())
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+
+
+
+    }
+
+    @And("^There is a game with incorrect id (\\d+)$")
+    public void ThereIsAGameWithIncorrectId(int id) throws Exception {
+        JSONObject game = new JSONObject();
+        game.put("id", id);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                get("/games/{id}",id)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(game.toString())
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+    @And("^I edit the pricePerCard to be (\\d+.\\d+) for the game with id (\\d+)$")
+    public void iEditThePricePerCardToBeForTheGameWithId(int newpricePerCard, int id) throws Exception {
+        JSONObject game = new JSONObject();
+        game.put("pricePerCard",newpricePerCard);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                put("/games/{id}", id)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(game.toString())
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+    @And("^The pricePerCard has been edited to be (\\d+.\\d+) for the game with id (\\d+)$")
+    public void thePricePerCardHasBeenEditedToBeForTheGameWithId(int pricePerCard, int id) throws Exception {
```

Comment:
The step claims to check that the pricePerCard has been edited but the corresponding parameter is not used at all